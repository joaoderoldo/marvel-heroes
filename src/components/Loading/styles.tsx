import styled from "styled-components";

export const LoadingContainer = styled.div`
  position: fixed;
  flex-flow: column wrap;
  display: flex;
  align-items: center;
  justify-content: center;
  width: 100%;
  height: 100vh;
  top: 0;
  left: 0;
  z-index: 1000;
  background-color: var(--secondary, gray);

  h1 {
    color: #707070;
    margin: 0;
    text-transform: uppercase;
    letter-spacing: 20px;
  }
`;
