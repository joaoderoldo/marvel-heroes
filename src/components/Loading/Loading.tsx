import { LoadingContainer } from "./styles";

import LoadingAnimation from "@/assets/gifs/heroe-loading.gif";

const Loading = () => {
  return (
    <LoadingContainer>
      <h1>Carregando</h1>
      <img src={LoadingAnimation} alt="Carregando" />
    </LoadingContainer>
  );
};

export default Loading;
