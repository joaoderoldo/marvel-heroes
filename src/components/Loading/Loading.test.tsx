import { render } from "@testing-library/react";
import Loading from "./Loading";

describe("Loading component", () => {
  it('renders the "Carregando" text', () => {
    const { getByText } = render(<Loading />);
    expect(getByText("Carregando")).toBeInTheDocument();
  });

  it("renders the loading animation", () => {
    const { getByAltText } = render(<Loading />);
    expect(getByAltText("Carregando")).toBeInTheDocument();
  });

  it("matches snapshot", () => {
    const { asFragment } = render(<Loading />);
    expect(asFragment()).toMatchSnapshot();
  });
});
