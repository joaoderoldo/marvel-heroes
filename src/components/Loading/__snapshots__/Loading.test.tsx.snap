// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Loading component > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-beqWaB kucQMd"
  >
    <h1>
      Carregando
    </h1>
    <img
      alt="Carregando"
      src="/src/assets/gifs/hero-loading.gif"
    />
  </div>
</DocumentFragment>
`;
